build/curriculum.ts

```diff
@@ -0,0 +1,422 @@
+    if (attributes.template === Template.module) {
+      prevNext = await buildPrevNextModule(slug);
+    } else if (attributes.template === Template.overview) {
+      prevNext = await buildPrevNextOverview(slug);
+    }
+
+    sidebar = await buildCurriculumSidebar();
+    parents = await buildParents(url);
+  } else {
+    title = title
+      .replace(/^\d+\s+/, "") // Strip number prefix.
+      .replace(/ modules$/, "") // Strip "modules" suffix.
+      .replace(/Extension \d+:/, ""); // Strip "Extension" prefix.
+  }
+
+  return {
+    meta: {
+      filename,
+      slug,
+      url,
+      title,
+      sidebar,
+      modules,
+      parents,
+      prevNext,
+      ...attributes,
+    },
+    body,
+  };
+}
+
+export async function findCurriculumPageBySlug(
+  slug: string
+): Promise<CurriculumData | null> {
+  let file = await slugToFile(slug);
+  if (!file) {
+    file = await slugToFile(`${slug}${slug ? "/" : ""}README`);
+  }
+  let module;
+  try {
+    module = await readCurriculumPage(file, { forIndex: false });
+  } catch (e) {
+    console.error(`No file found for ${slug}: ${e}`);
+    return;
+  }
+  const { body, meta } = module;
+
+  const d: CurriculumBuildData = {
+    url: meta.url,
+    rawBody: body,
+    metadata: { locale: DEFAULT_LOCALE, ...meta },
+    isMarkdown: true,
+    fileInfo: {
+      path: file,
+    },
+  };
+
+  const doc = await buildCurriculumPage(d);
+  return { doc };
+}
+
+export async function buildCurriculumPage(
+  document: CurriculumBuildData
+): Promise<Doc> {
+  const { metadata } = document;
+
+  const doc = { locale: DEFAULT_LOCALE } as Partial<CurriculumDoc>;
+  let $ = null;
+
+  const renderUrl = document.url.replace(/\/$/, "");
+  [$] = await kumascript.render(renderUrl, {}, document as any);
+
+  $("[data-token]").removeAttr("data-token");
+  $("[data-flaw-src]").removeAttr("data-flaw-src");
+
+  doc.title = (metadata.title || "").replace(/^\d+\s+/, "");
+  doc.mdn_url = document.url;
+  doc.locale = metadata.locale as string;
+  doc.native = LANGUAGES_RAW[DEFAULT_LOCALE]?.native;
+
+  if ($("math").length > 0) {
+    doc.hasMathML = true;
+  }
+  $("div.hidden").remove();
+  syntaxHighlight($, doc);
+  injectNoTranslate($);
+  injectLoadingLazyAttributes($);
+  postProcessCurriculumLinks($, (p: string) => {
+    const [head, hash] = p.split("#");
+    const slug = fileToSlug(
+      path.normalize(path.join(path.dirname(document.fileInfo.path), head))
+    ).replace(/\/$/, "");
+    return `/${DEFAULT_LOCALE}/${slug}/${hash ? `#${hash}` : ""}`;
+  });
+  postProcessExternalLinks($);
+  postLocalFileLinks($, doc);
+  postProcessSmallerHeadingIDs($);
+  wrapTables($);
+  setCurriculumTypes($);
+  try {
+    const [sections] = await extractSections($);
+    doc.body = sections;
+  } catch (error) {
+    console.error(
+      `Extracting sections failed in ${doc.mdn_url} (${document.fileInfo.path})`
+    );
+    throw error;
+  }
+
+  doc.pageTitle = `${doc.title} | MDN Curriculum`;
```

Comment:
```suggestion
  doc.pageTitle = doc.title ? `${doc.title} | MDN Curriculum` : "MDN Curriculum";
```

Even better: Reuse `TITLE_SUFFIX` (but this is currently only defined in `client`).